src/transformers/trainer.py

```diff
@@ -1983,7 +1983,9 @@ def _inner_training_loop(
                     # if loss is nan or inf simply add the average of previous logged losses
                     tr_loss += tr_loss / (1 + self.state.global_step - self._globalstep_last_logged)
                 else:
-                    assert tr_loss.device == tr_loss_step.device, f"Calculated loss must be on the original device: {tr_loss.device} but device in use is {tr_loss_step.device}"
+                    assert (
+                        tr_loss.device == tr_loss_step.device
+                    ), f"Calculated loss must be on the original device: {tr_loss.device} but device in use is {tr_loss_step.device}"
```

Comment:
please verify the `x.device` is printable so it will have a message that makes sense and not a memory address of the object :)

Reply:
Confirmed. tested. it's  printable.